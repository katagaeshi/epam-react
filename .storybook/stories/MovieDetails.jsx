import React from 'react';
import {
  Provider,
  connect,
} from 'react-redux';
import { storiesOf } from '@storybook/react';

import configureStore from '../../public/store';

import MovieDetails from '../../public/components/containers/MovieDetails';
import RunningTitle from '../../public/components/ui/RunningTitle';

import storeInitialState from '../../public/store/initialState.json';

const { store } = configureStore(storeInitialState);

storiesOf('MovieDetails', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('Example of movie', () => (
      <div>
        <RunningTitle/>
        <MovieDetails
          poster_path='https://image.tmdb.org/t/p/w500/jBibISmRBUO35p0Picv0nOo33sc.jpg'
          title='Title'
          vote_average='7.3'
          release_date='2014-01-05'
          tagline='Trust No One'
          runtime='168'
          overview='Jack Ryan, as a young covert CIA analyst, uncovers a Russian plot to crash the U.S. economy with a terrorist attack.'
          onClick={() => {}}
          notFound={false}
          location={{pathname:''}}
          match={{path:''}}
        />
        <RunningTitle/>
      </div>
    )
  );
