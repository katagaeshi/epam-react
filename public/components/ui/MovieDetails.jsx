import React from 'react';
import PropTypes from 'prop-types';

const MovieDetails = props => (
  <div>
    <button>SEARCH</button>
    <img src={props.src} alt="poster" />
    <span>{props.title}</span>
    <span>{props.rating}</span>
    <span>{props.tag}</span>
    <span>{props.year}</span>
    <span>{props.duration}</span>
    <span>{props.description}</span>
  </div>
);

MovieDetails.propTypes = {
  src: PropTypes.string,
  title: PropTypes.string.isRequired,
  rating: PropTypes.string,
  year: PropTypes.number,
  tag: PropTypes.string,
  duration: PropTypes.string,
  description: PropTypes.string,
};

MovieDetails.defaultProps = {
  src: null,
  year: null,
  tag: null,
  duration: null,
  description: null,
  rating: null,
};

export default MovieDetails;
