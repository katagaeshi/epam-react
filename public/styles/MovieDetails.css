.MovieDetails {
  display: grid;
  background-color: gray;
  grid-auto-columns: 3fr 1fr 1fr 2fr;
  grid-auto-rows: 0.2fr 0.3fr 0.2fr 2fr;
}

.MovieDetails > button {
  height: 30px;
  width: 100px;
  position: absolute;
  top: 20px;
  right: 20px;
}

.MovieDetails > img {
  grid-row: 1 / 5;
  grid-column: 1 / 1;
  max-width: 100%;
}

.MovieDetails > .Title {
  grid-row: 1 / 1;
  grid-column: 2 / 2;
  font-weight: bold;
  font-size: 20px;
  color: pink;
}

.MovieDetails > .VoteAverage {
  grid-row: 1 / 1;
  grid-column: 3 / 3;
  color: white;
  border-radius: 50%;
  border: solid white;
  width: 30px;
  height: 25px;
  text-align: center;
  vertical-align: middle;
  padding-top: 5px;
}

.MovieDetails > .TagLine {
  grid-row: 2 / 2;
  grid-column: 2 / 3;
  color: white;
}

.MovieDetails > .ReleaseDate {
  grid-row: 3 / 3;
  grid-column: 2 / 2;
  color: white;
}

.MovieDetails > .Runtime {
  grid-row: 3 / 3;
  grid-column: 3 / 3;
  color: white;
}

.MovieDetails > .Overview {
  grid-row: 4 / 4;
  grid-column: 2 / 4  ;
  color: white;
}
  
