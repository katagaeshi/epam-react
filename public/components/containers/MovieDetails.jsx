import { connect } from 'react-redux';
import MovieDetails from '../ui/MovieDetails';
import { setActivePanel } from '../../actions';

const mapStateToProps = state => state.foundMovies.value.movies[state.movieDetails];

const mapDispatchToProps = dispatch => ({
  onClick() {
    dispatch(setActivePanel('SearchPanel'));
  },
});

const Container = connect(mapStateToProps, mapDispatchToProps)(MovieDetails);
export default Container;
