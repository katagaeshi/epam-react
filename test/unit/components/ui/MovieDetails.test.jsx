import MovieDetails from 'public/components/ui/MovieDetails';

describe('MovieDetails', () => {
  it('Snapshot', () => {
    const tree = shallow(<MovieDetails
      poster_path="www.example.com"
      title="Jay and Pikachu Strike Back"
      vote_average="6.9"
      release_date="a long time ago"
      tagline="movieForNerds"
      runtime="01:31"
      overview="Probably the most popular anime in the world"
    />);
    expect(tree).toMatchSnapshot();
  });
});
