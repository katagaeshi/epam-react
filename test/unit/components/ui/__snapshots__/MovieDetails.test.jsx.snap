// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails Snapshot 1`] = `
<MovieDetails
  classes={
    Object {
      "button": "MovieDetails-button-0-1-2",
      "image": "MovieDetails-image-0-1-3",
      "movieDetails": "MovieDetails-movieDetails-0-1-1",
      "overview": "MovieDetails-overview-0-1-10",
      "releaseDate": "MovieDetails-releaseDate-0-1-8",
      "runtime": "MovieDetails-runtime-0-1-9",
      "tagLine": "MovieDetails-tagLine-0-1-7",
      "title": "MovieDetails-title-0-1-5",
      "titleAndVote": "MovieDetails-titleAndVote-0-1-4",
      "voteAverage": "MovieDetails-voteAverage-0-1-6",
    }
  }
  notFound={false}
  overview="Probably the most popular anime in the world"
  poster_path="www.example.com"
  release_date="a long time ago"
  runtime="01:31"
  tagline="movieForNerds"
  title="Jay and Pikachu Strike Back"
  vote_average="6.9"
/>
`;
